kalpsdk/read_transaction.go

```diff
@@ -6,11 +6,77 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"time"
 
 	//Third party Libs
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+// IsSmartContractOwner checks if the caller is the owner of the smart contract
+// by comparing the userID of the caller with the stored smart contract owner ID.
+//
+// Returns:
+//   - bool: A boolean value indicating whether the caller is the owner of the smart contract.
+//   - error: An error if the operation fails.
+func (ctx *TransactionContext) IsSmartContractOwner() (bool, error) {
+	userID, err := ctx.GetUserID()
+	if err != nil {
+		return false, err
+	}
+
+	ownerId, err := ctx.GetStub().GetState("smartContractOwner")
+	if err != nil {
+		return false, err
+	}
+
+	return string(ownerId) == userID, nil
+}
+
+// FetchOwnerHistory retrieves the transaction history of the smart contract owner.
+//
+// Returns:
+//   - []HistoryQueryResult: A slice containing the transaction history of the smart contract owner.
+//   - error: An error if the operation fails.
+func (ctx *TransactionContext) FetchOwnerHistory() ([]HistoryQueryResult, error) {
+	isOwner, err := ctx.IsSmartContractOwner()
+	if err != nil {
+		return nil, err
+	}
+	if !isOwner {
+		return nil, fmt.Errorf("signer is not an owner")
+	}
```

Comment:
Why there are 2 checks, cant we handle it with if condition?

Reply:
No, we can't handle this because the error for not an owner is a custom error. that's why I have to put 2 conditions.